developers/weaviate/client-libraries/typescript/typescript-v3.md

```diff
@@ -10,91 +10,93 @@ import Tabs from '@theme/Tabs';
+:::
+
+For usage information not specific to the TypeScript client, such as code examples, see the relevant pages in the Weaviate documentation.
+
+## Client Setup
+
+### Install
+
+Use [npm](https://www.npmjs.com/) to install the TypeScript client library package:
 
 ```bash
 npm install weaviate-client --tag beta
 ```
 
-Some packages, like the Weaviate TypeScript client, require extra configuration. The root directory of a TypeScript project has a `tsconfig.json` file. Add the following to your `tsconfig.json`. 
+### Import the Client
 
-<details>
-    <summary> tsconfig.json file</summary>
-    To properly use the client, add the following to your tsconfig.json file:
-
-    ```json
-       {
-            "compilerOptions": {
-              ...
-              "target": "esnext",
-              "module": "esnext", 
-              "moduleResolution": "Node16",
-              "include": ["*.ts"], 
-              "esModuleInterop": true,
-              "lib": [ "es2018" ],
-              ...
-          }
-        }
-    ```
-    
-</details>
+The `v3` client uses `ES Modules`. Most of the sample code follows `ES Module` style.
 
-Don't specify filenames on the command line when you use `tsconfig.json`. Specify the TypeScript files in `tsconfig.json` instead. `tsc` only reads `tsconfig.json` when you run it by itself.
+If your code requires `CommonJS` compatibility, use the `CommonJS` import style:
 
+<Tabs groupId="languages">
+<TabItem value="esm" label="ES Modules">
 
-## References
+```ts
+import weaviate from 'weaviate-client'
+```
 
-:::note v2 → v3 client documentation migration
-We are in the process of migrating our v2 client code examples to v3 as necessary. Please be patient as we work on documenting the rest.
-:::
+</TabItem>
+<TabItem value="cjs" label="CommonJS">
 
-At the moment, we've covered the following pages. 
+```ts
+const weaviate = require('weaviate-client').default;
+```
 
-- [Introduction](https://weaviate.io/developers/weaviate/introduction)
-- [Quickstart](https://weaviate.io/developers/weaviate/quickstart)
-- [Start Guides](https://weaviate.io/developers/weaviate/starter-guides)
-- [How-to: Search](https://weaviate.io/developers/weaviate/search)
-- [How-to: Manage Data](https://weaviate.io/developers/weaviate/manage-data)
+</TabItem>
+</Tabs>
 
 
-Our [RESTful endpoints](../../api/rest/index.md) and [GraphQL functions](../../api/graphql/index.md) covered by the TypeScript client currently have JavaScript examples in the code blocks.
+### Typescript Setup
 
-## Design
+Add the following code to your `tsconfig.json`file. 
 
-### A Collection first Approach
+<details>
+    <summary>tsconfig.json file</summary>
+Change or add the following properties.
 
-With the new v3 client, we've opted focus on collections as primary method of interaction with your Weaviate database. We've also utilized strong typing through custom Typescript types and user-defined generics.
+```json
+{
+    "compilerOptions": {
```

Comment:
Reminder re: our discussion - let's make sure that this works with TS top level awaits not wrapped in `main`

Reply:
Add: "type": "module" to `package.json`

Reply:
e.g.:

```json
{
  "dependencies": {
    "weaviate-client": "^3.0.0-beta.23"
  },
  "type": "module"
}
```

Reply:
added!